locking/locks.go

```diff
@@ -225,33 +249,101 @@ func (c *Client) lockIdFromPath(path string) (string, error) {
 	}
 }
 
+// We want to use a single cache file for integrity, but to make it easy to
+// list all locks, prefix the id->path map in a way we can identify (something
+// that won't be in a path)
+const idKeyPrefix = "*id*://"
+
+func (c *Client) encodeIdKey(id string) string {
```

Comment:
I like this approach a lot. I think it's a nice, clear line of abstraction that we can draw. The `*locking.Client` would own a `*LockCache` underneath and delegate into the cache's public methods from there.